infrastructure/templates/public-api/components/alerts/appServicePlans/cpuPercentageAlert.bicep

```diff
@@ -0,0 +1,19 @@
+@description('Names of the resources that these alerts are being applied to.')
+param resourceNames string[]
+
+@description('Name of the Alerts Group used to send alert messages.')
+param alertsGroupName string
+
+@description('Tags with which to tag the resource in Azure.')
+param tagValues object
+
+module alerts '../baseCpuPercentageAlert.bicep' = {
```

Comment:
Are we potentially over-abstracting by having loads of extra modules that wrap `baseCpuPercentageAlert.bicep`?

It's not clear what extra value we're getting from the additional wrapping and feels like we're just copy-pasting the same set of parameters and configuration in a bunch of different modules. It also looks we're generating a bunch of additional resource group deployments due to the extra wrapping, which is a little less clear to debug.

I'd potentially suggest we simplify things and remove the modules that are re-using the 'base' modules like `baseCpuPercentageAlert.bicep`. Instead, we can just use the base modules directly and avoid the extra abstraction layer.

Similar thing would apply across this PR.